savant/utils/sink_factories.py

```diff
@@ -99,6 +100,10 @@ def __init__(
         socket_type: str = SenderSocketTypes.PUB.name,
         bind: bool = True,
         send_hwm: int = Defaults.SEND_HWM,
+        receive_timeout: int = Defaults.SENDER_RECEIVE_TIMEOUT,
+        # TODO: Pipeline hangs on shutdown when zmq.Again raised.
```

Comment:
Not clear what this TODO is about? What does it mean to hang and change the value?

Reply:
When sink_factory raise an error `zmq.Again` pipeline hangs on shutdown. This can happen when socket is `req+connect`.

Reply:
@tomskikh how can we handle it to avoid the hang?

Reply:
@bwsw I tried to add `exit(1)` on this error, but this didn't work.

```python
        with NvDsPipelineRunner(pipeline):
            try:
                for msg in pipeline.stream():
                    sink(msg, **dict(module_name=config.name))
            except Exception as exc:  # pylint: disable=broad-except
                logger.error('Error: %s',exc, exc_info=True)
                exit(1)
```

Maybe we need to drain the queue (`pipeline.stream()`) when shutting down the pipeline. I suggest to keep this workaround (`req_receive_retries: int =  1000`) for now and create a separate issue to fix this.

Reply:
Consider: 
https://stackoverflow.com/questions/905189/why-does-sys-exit-not-exit-when-called-inside-a-thread-in-python

Reply:
Added `os._exit(1)`.